OpenRA.Mods.Common/Traits/World/EditorCursorLayer.cs

```diff
@@ -189,6 +189,22 @@ public int SetActor(WorldRenderer wr, ActorInfo actor, PlayerReference owner)
 			return ++CurrentToken;
 		}
 
+		public void MoveActor(EditorActorPreview preview, CPos location)
+		{
+			editorLayer.Remove(preview);
```

Comment:
I think removing is wrong. We should instead add an update function to EditorActorLayer

Reply:
I'm giving this a go but it is proving to be a bit difficult. It makes a lot of sense to remove and recreate because lots of items like the footprint, LocationInit, and RuntimeNeighbourInit need to be removed and recreated, too. A dedicated Update function essentially recreates a lot of stuff that the Remove/Add functions are already doing and only really increases future maintenance burden.
However, I am recovering from Covid as I write this and might not be in my right state of mind.

Reply:
We also remove and add actors to the world instead of updating them, so this could be fine.